src/templates/TemplateData.ts

```diff
@@ -147,29 +168,37 @@ export async function tryGetTemplateDataFromCache(reporter: TelemetryReporter |
     }
 }
 
-export async function tryGetTemplateDataFromFuncPortal(reporter: TelemetryReporter | undefined, globalState?: vscode.Memento, hostname: string = 'functions.azure.com'): Promise<TemplateData | undefined> {
+export async function tryGetTemplateDataFromFuncPortal(reporter: TelemetryReporter | undefined, globalState?: vscode.Memento): Promise<TemplateData | undefined> {
     try {
         return <TemplateData>await callWithTelemetryAndErrorHandling('azureFunctions.tryGetTemplateDataFromFuncPortal', reporter, undefined, async function (this: IActionContext): Promise<TemplateData> {
+            if (await cachedTemplatesAreCurrent(globalState)) {
+                throw new Error('Cached templates are current-- no need');
```

Comment:
This shouldn't be an error. It will mess up our telemetry numbers and we don't want to risk accidentally showing this to the user. You'll have to refactor a bit, but you should probably just return undefined